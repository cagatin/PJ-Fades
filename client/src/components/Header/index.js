import React from 'react'

import Container from 'react-bootstrap/Container';
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';

import logo from '../../images/logo/logo.png'

function Header(props) {

  const style = {
    container: {
        margin: "0",
        maxWidth: "100%",
        fontColor: "white",
    },
    image: {
        width: "92px",
        height: "75px"
    },
    nav: {
        width: "60%",
        marginRight: "10%",
        fontSize: "20px"
    }
  }

  return (
    <Navbar className="fixed-top" bg="light" expand="sm">
      <Container style={style.container}>
        <Navbar.Brand href="/">𝐏𝐉-𝐅𝐀𝐃𝐄𝐒</Navbar.Brand>
        <img src={logo} alt="logo" style={style.image}></img>
        <Navbar.Toggle aria-controls="navbar" />
        <Container fluid="lg" style={style.container}> 
            <Navbar.Collapse className="justify-content-end" id="navbar">
            <Nav className="justify-content-around" style={style.nav}>
                <Nav.Link href="/">Home</Nav.Link>
                <Nav.Link href="/gallery">Gallery</Nav.Link>
                <Nav.Link href="/contact">Contact</Nav.Link>
                <Nav.Link href="/reservation">Reservation</Nav.Link>
            </Nav>
            </Navbar.Collapse>
        </Container>
      </Container>
    </Navbar>
)
}
export default Header;