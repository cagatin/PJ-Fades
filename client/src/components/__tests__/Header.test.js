import React from 'react';
import { createRoot } from 'react-dom/client';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import Header from '../Header';

let container;
let root;
globalThis.IS_REACT_ACT_ENVIRONMENT = true;

beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    act(() => {
        root = createRoot(container);
    })
});

afterEach(() => {
    act(() => {
        root.unmount();
    })
    container.remove();
    container = null;
});

describe('Header', () => {
    it('should match snapshot', () => {
        act(() => {
            root.render(<Header />);
        });
        expect(pretty(container.innerHTML)).toMatchSnapshot();
    });
});