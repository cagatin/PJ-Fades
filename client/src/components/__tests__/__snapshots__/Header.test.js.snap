// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
"<nav class=\\"fixed-top navbar navbar-expand-sm navbar-light bg-light\\">
  <div style=\\"margin: 0px; max-width: 100%;\\" class=\\"container\\"><a href=\\"/\\" class=\\"navbar-brand\\">𝐏𝐉-𝐅𝐀𝐃𝐄𝐒</a><img src=\\"logo.png\\" alt=\\"logo\\" style=\\"width: 92px; height: 75px;\\"><button aria-controls=\\"navbar\\" type=\\"button\\" aria-label=\\"Toggle navigation\\" class=\\"navbar-toggler collapsed\\"><span class=\\"navbar-toggler-icon\\"></span></button>
    <div style=\\"margin: 0px; max-width: 100%;\\" class=\\"container-lg\\">
      <div class=\\"justify-content-end navbar-collapse collapse\\" id=\\"navbar\\">
        <div class=\\"justify-content-around navbar-nav\\" style=\\"width: 60%; margin-right: 10%; font-size: 20px;\\"><a href=\\"/\\" data-rr-ui-event-key=\\"/\\" class=\\"nav-link\\">Home</a><a href=\\"/gallery\\" data-rr-ui-event-key=\\"/gallery\\" class=\\"nav-link\\">Gallery</a><a href=\\"/contact\\" data-rr-ui-event-key=\\"/contact\\" class=\\"nav-link\\">Contact</a><a href=\\"/reservation\\" data-rr-ui-event-key=\\"/reservation\\" class=\\"nav-link\\">Reservation</a><a href=\\"/login\\" data-rr-ui-event-key=\\"/login\\" class=\\"nav-link\\">Login</a></div>
      </div>
    </div>
  </div>
</nav>"
`;
